cartographer/mapping/sparse_pose_graph.h

```diff
@@ -120,6 +125,9 @@ class SparsePoseGraph {
   virtual std::vector<Constraint> constraints() = 0;
 };
 
+SparsePoseGraph::Constraint::Tag FromProto(
```

Comment:
How about exposing instead a function that creates a vector of `SparsePoseGraph::Constraint` from the proto?

Reply:
In other words, you want to extract [lines 433-446](https://github.com/googlecartographer/cartographer/pull/544/files#diff-8148ee0da898a26c8b927bc8079d8900R433) into a separate function?

Reply:
Yes. I find it always hard to read overly long closures.

Reply:
Creating a constraint is not possible to express as a pure function. It requires access to per-node `gravity_alignment` for 2D (see line [442](https://github.com/googlecartographer/cartographer/pull/544/files#diff-8148ee0da898a26c8b927bc8079d8900R442)).